### PR TITLE
feat: #348 - new methods getTaxonomyTranslationUri and getCrowdinUri

### PR DESCRIPTION
**Pull request by monsieurtanuki**
@teolemon the URI does not work for EMB_CODES

Impacted files:
* `api_getProduct_test.dart`: tests for new methods `getTaxonomyTranslationUri` and `getCrowdinUri`
* `openfoodfacts.dart`: new methods `getTaxonomyTranslationUri` and `getCrowdinUri`

### CODE REVIEW

**Comment by teolemon** [comment]
Traceability codes (how I renamed the superset of EMB_CODES) is normalized by replacing everything that looks like a point or a space by -
https://world.openfoodfacts.org/packager-codes
Not sure if the API returns a normalized version.
https://world.openfoodfacts.org/packager-code/fr-72-264-002-ce
https://world.openfoodfacts.org/packager-code/fr-72-264-002-ce.json

**Comment by teolemon** [comment]
- Already the Taxonomy & Crowdin are great. 
- Eventually, being able to show facets in the apps (like emb_codes) would be nice, but it might be another PR

File: lib/openfoodfacts.dart [hunk 1]
```diff
@@ -244,6 +244,37 @@ class OpenFoodAPIClient {
...
+    return UriHelper.replaceSubdomain(
+      uri,
+      language: language,
+      country: country,
+    );
+  }
+
+  /// Returns the URI to the crowdin page for a [language].
+  static Uri getCrowdinUri(final OpenFoodFactsLanguage language) =>
+      Uri.parse('https://crowdin.com/project/openfoodfacts/${language.code}');
```
Comment (M123-dev) [inline lib/openfoodfacts.dart:276]: Does this also work for GB and EN

File: lib/openfoodfacts.dart [hunk 2]
```diff
@@ -244,6 +244,37 @@ class OpenFoodAPIClient {
...
+  /// Returns the URI to the translation page for a taxonomy.
+  ///
+  /// If the target website supports different domains for country + language,
+  /// [replaceSubdomain] should be set to true.
+  static Uri getTaxonomyTranslationUri(
+    final TagType taxonomyTagType, {
+    required final OpenFoodFactsLanguage language,
+    final OpenFoodFactsCountry? country,
+    final QueryType? queryType,
+    required final bool replaceSubdomain,
```
Comment (M123-dev) [inline lib/openfoodfacts.dart:256]: What about giving it an initial value and not requiring it

**Review by M123-dev — APPROVED** [review]

**Comment by monsieurtanuki** [comment]
Thank you @M123-dev and @teolemon for your review!
* Regarding EMB_CODES, now I throw an exception in `getTaxonomyTranslation` because there's no needed translation for them (but as it's a `TagType` "taxonomy" it can be an input parameter)
* Regarding setting the country for `getTaxonomyTranslation`, it does not seem to make much sense - I removed that parameter (and therefore it will always be `world-$countryCode`)

**Pull request merged**